internal/proxysql/core.go

```diff
@@ -1,160 +1,245 @@
 package proxysql
 
 import (
-	"context"
 	"fmt"
 	"log/slog"
 	"os"
-	"sort"
-	"strings"
 	"time"
 
-	"github.com/persona-id/proxysql-agent/internal/configuration"
+	// fuck outta here.
+	_ "github.com/go-sql-driver/mysql"
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
 )
 
+// ProxySQL core functions.
 //
-// Core mode specific settings
+// The core pods need to run certain commands when specific pods joins or leaves the
+// cluster, so this function sets up an informer that watches the k8s pods and runs
+// functions when pods change.
 //
+// Joining:
+//
+// When a new core pod joins the cluster, one of two things happen:
+//   - if it's the first core pod, it uses the podAdded callback to add itself to the proxysql_servers table
+//   - if other core pods are already running, one of them will use add the new pod via the podUpdated function
+//
+// When a new satellite pod joins the cluster, the core pods all run the "LOAD X TO RUNTIME" commands, which
+// accepts the new pod and distributes the configuration to it.
+//
+// Leaving:
+//
+//   - When a satellite pod leaves the cluster, nothing needs to be done.
+//   - When a core pod leaves the cluster, the remaining core pods all delete that pod from the proxysql_servers
+//     table and run all of the LOAD X TO RUNTIME commands.
+func (p *ProxySQL) Core() {
+	if p.clientset == nil {
+		config, err := rest.InClusterConfig()
+		if err != nil {
+			slog.Error("error", slog.Any("err", err))
```

Comment:
Probably want to bail here and after `kubernetes.NewForConfig` if there was an error.